javaServices/coreJavaServices/src/main/java/joliex/db/DatabaseService.java

```diff
@@ -115,148 +128,102 @@ public void connect( Value request )
 				}
-			} else {
-				Class.forName( driverClass );
 			}
+			Class.forName( driverClass );
 
 			if( isEmbedded ) {
 				connectionString = "jdbc:" + driver + ":" + databaseName;
 				if( !attributes.isEmpty() ) {
 					connectionString += ";" + attributes;
 				}
-				if( "hsqldb".equals( driver ) ) {
-					connection = DriverManager.getConnection(
-						connectionString,
-						username,
-						password );
-				} else {
-					connection = DriverManager.getConnection( connectionString );
+				connectionPool = _createDataSource();
+				if( !"hsqldb".equals( driver ) ) { // driver == sqlite || driver == derby_embedded
+					connectionPool.setUsername( null );
+					connectionPool.setPassword( null );
 				}
 			} else {
 				if( driver.startsWith( "hsqldb" ) ) {
 					connectionString = "jdbc:" + driver + ":" + driver.substring( driver.indexOf( '_' ) + 1 ) + "//"
 						+ host + (port.isEmpty() ? "" : ":" + port) + separator + databaseName;
 				} else {
-					connectionString =
-						"jdbc:" + driver + "://" + host + (port.isEmpty() ? "" : ":" + port) + separator + databaseName;
+					connectionString = "jdbc:" + driver + "://" + host + (port.isEmpty() ? "" : ":" + port) + separator
+						+ databaseName;
 				}
 				if( encoding.isPresent() ) {
 					connectionString += "?characterEncoding=" + encoding.get();
 				}
-				connection = DriverManager.getConnection(
-					connectionString,
-					username,
-					password );
+				connectionPool = _createDataSource();
 			}
 
-			if( connection == null ) {
-				throw new FaultException( "ConnectionError" );
-			} else {
-				final Connection currConnection = connection;
-				interpreter().cleaner().register( this, () -> {
-					try {
-						if( !currConnection.isClosed() ) {
-							currConnection.close();
-						}
-					} catch( SQLException e ) {
-					}
-				} );
-			}
+			interpreter().cleaner().register( this, () -> {
+				connectionPool.close();
+			} );
 		} catch( ClassNotFoundException e ) {
 			throw new FaultException( "DriverClassNotFound", e );
-		} catch( SQLException e ) {
-			throw createFaultException( e );
 		}
 	}
 
 	private void _checkConnection()
 		throws FaultException {
-		if( connection == null ) {
+		if( connectionPool == null ) {
 			throw new FaultException( "ConnectionError" );
 		}
 		if( mustCheckConnection ) {
-			try {
-				if( "postgresql".equals( driver ) ) {
-					/*
-					 * The JDBC4 driver for postgresql does not implemented isValid(). We fallback to isClosed().
-					 */
-
-					if( connection.isClosed() ) {
-						connection = DriverManager.getConnection(
-							connectionString,
-							username,
-							password );
-					}
-				} else {
-					if( !connection.isValid( 0 ) ) {
-						connection = DriverManager.getConnection(
-							connectionString,
-							username,
-							password );
-					}
-				}
-			} catch( SQLException e ) {
-				throw createFaultException( e );
+			if( connectionPool.isClosed() ) {
```

Comment:
HikariCp will validate connections before handing them out, so it suffices to check whether the connection pool is closed.